MapleServer2/PacketHandlers/Game/RequestItemUseMultipleHandler.cs

```diff
@@ -39,43 +40,27 @@ public override void Handle(GameSession session, PacketReader packet)
                 }
             }
 
-            short opened = 0; // Amount of opened boxes
-            List<Item> items = new List<Item>(session.Player.Inventory.Items.Values); // Make copy of items in-case new item is added
+            Dictionary<long, Item> items = new Dictionary<long, Item>(session.Player.Inventory.Items.Where(x => x.Value.Id == boxId)); // Make copy of items in-case new item is added
 
-            foreach (Item item in items)
+            foreach (KeyValuePair<long, Item> kvp in items)
             {
-                // Continue over non-matching item ids
-                if (item.Id != boxId)
-                {
-                    continue;
-                }
-
+                Item item = kvp.Value;
                 // Do nothing if box has no data stored
                 if (item.Content.Count <= 0)
                 {
                     break;
                 }
 
-                for (int i = opened; i < amount; i++)
+                for (int i = 0; i < amount; i++)
```

Comment:
This is incorrect, without keeping track of number of opened boxes it'll keep resetting the amount of boxes to open when it goes to the next stack of boxes.